sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogRecordBuilder.java

```diff
@@ -95,6 +109,14 @@ public void emit() {
       return;
     }
     Context context = this.context == null ? Context.current() : this.context;
+    long timestampEpochNanos =
+        this.timestampEpochNanos == 0
+            ? this.loggerSharedState.getClock().now()
+            : this.timestampEpochNanos;
+    long observedTimestampEpochNanos =
+        this.observedTimestampEpochNanos == 0
+            ? timestampEpochNanos
+            : this.observedTimestampEpochNanos;
```

Comment:
Let's pause on merging this until this [convo](https://github.com/open-telemetry/opentelemetry-specification/pull/3385#discussion_r1164519850) is resolved. There is some disagreement about what to do if observed timestamp is not explicitly set. We could set it equal to timestamp (as shown in this PR) or set it to the current time.